src/services/supabase.ts

```diff
@@ -132,6 +133,47 @@ export const defaultTableFilter = <Data>(
     return queryBuilder;
 };
 
+export const distributedTableFilter = <Data>(
```

Comment:
This is a direct copy of `defaultTableFilter` above, with the exception of the distributed type annotation applied to the `query` input parameter. In a fast follow, we should look at consolidating the two functions.